### ISSUE TITLE
Is there data augmentation probability for rotation ?

### ISSUE THREAD

**Issue by Mitix-EPI** [ultralytics/ultralytics#8380]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi !
I trained a model on my custom data and found out that when I rotate my object, it's recognize with high confidence. But when i put my object strictly straight on, the model does'nt recognize the object with hight confidence (0.3-0.6).
I have thousands of data with my object strictly straight on. So my hypothesis is that the rotating random augmention during training fully transforms all my data and can't recognize "normal" image. 
I have read the documentation and i did not find a way to have random rotation with probability. 

Is there any hidden hyperparameter that I can switch on ?

**Comment by glenn-jocher** [comment]
@Mitix-EPI hi there! 👋

In YOLOv8, data augmentation is applied by default, which includes random rotations. However, there isn't a specific hyperparameter to adjust the probability of rotation directly. If you suspect that the augmentation is too strong for your dataset, you might consider customizing the augmentation settings.

You can modify the augmentation parameters in the `data.yaml` file associated with your dataset. Look for the `augment` section and adjust the `degrees` parameter to control the rotation range. Setting it to a lower value will reduce the extent of rotation applied during training.

Here's a quick example of how you might adjust it:

```yaml
# data.yaml
train: ...
val: ...

# Augmentation settings
augment:
  degrees: 5  # Lower this value to reduce rotation range
  translate: 0.1
  scale: 0.5
  shear: 0.1
  perspective: 0.0
  flipud: 0.0
  fliplr: 0.5
  mosaic: 1.0
  mixup: 0.0
```

Hope this helps! Let us know if you have any more questions. 😊

**Comment by Burhan-Q** [comment]
@Mitix-EPI you're right, this [Train Argument is missing from the docs page](https://docs.ultralytics.com/modes/train/#arguments). As @glenn-jocher mentioned, you can include `degrees=R` where `R` is the max value of rotation (+/-) to randomly apply to images during training.

```py
from ultralytics import YOLO

model = YOLO("yolov8n.pt")

results = model.train(
    data="path/to/data.yaml",
    ..., # other training arguments
    degrees=45, # applies max +/- 45-degree rotation randomly
)
```

**Comment by Burhan-Q** [comment]
- [ ] add `degrees` training argument to the Modes / Train documentation page

**Comment by Burhan-Q** [comment]
Fixed in #8415
